pragma-node/src/infra/repositories/onchain_repository/history.rs

```diff
@@ -70,3 +103,175 @@ pub async fn get_historical_aggregated_entries(
+            )
+            .await?;
+
+            if base_alt_result.0.len() != alt_quote_result.0.len() {
+                continue;
+            }
+
+            return calculate_rebased_prices(base_alt_result, alt_quote_result);
+        }
+    }
+
+    Err(InfraError::RoutingError)
+}
+
+/// Given two vector of entries, compute a new vector containing the routed prices.
+fn calculate_rebased_prices(
+    base_result: (Vec<HistoricalEntryRaw>, u32),
+    quote_result: (Vec<HistoricalEntryRaw>, u32),
+) -> Result<(Vec<HistoricalEntryRaw>, u32), InfraError> {
+    let (base_entries, base_decimals) = base_result;
+    let (quote_entries, quote_decimals) = quote_result;
+
+    let (rebased_entries, decimals) = if base_decimals < quote_decimals {
+        let normalized_base_entries =
+            normalize_entries_to_decimals(base_entries, base_decimals, quote_decimals);
+        (
+            convert_entries_via_quote(normalized_base_entries, quote_entries, quote_decimals)?,
+            quote_decimals,
+        )
+    } else {
+        let normalized_quote_entries =
+            normalize_entries_to_decimals(quote_entries, quote_decimals, base_decimals);
+        (
+            convert_entries_via_quote(base_entries, normalized_quote_entries, base_decimals)?,
+            base_decimals,
+        )
+    };
+
+    Ok((rebased_entries, decimals))
+}
+
+/// Wrapper around the [normalize_to_decimals] function. It will be applied to all
+/// entries median_price.
+fn normalize_entries_to_decimals(
+    entries: Vec<HistoricalEntryRaw>,
+    base_decimals: u32,
+    quote_decimals: u32,
+) -> Vec<HistoricalEntryRaw> {
+    entries
+        .into_iter()
+        .map(|mut entry| {
+            entry.median_price =
+                normalize_to_decimals(entry.median_price, base_decimals, quote_decimals);
+            entry
+        })
+        .collect()
+}
+
+/// Wrapper around the [convert_via_quote] function. It will be applied to all
+/// normalized_entries median_price.
+fn convert_entries_via_quote(
+    normalized_entries: Vec<HistoricalEntryRaw>,
+    quote_entries: Vec<HistoricalEntryRaw>,
+    decimals: u32,
+) -> Result<Vec<HistoricalEntryRaw>, InfraError> {
+    normalized_entries
+        .into_iter()
+        .zip(quote_entries)
+        .map(|(base_entry, quote_entry)| {
+            let converted_price = convert_via_quote(
+                base_entry.median_price.clone(),
+                quote_entry.median_price.clone(),
+                decimals,
+            )?;
+            combine_entries(&base_entry, &quote_entry, converted_price)
+        })
+        .collect()
+}
+
+/// Given two entries, determine what should be the resulted pair_id, timestamp
+/// & sources. Returns the new entry afterwards.
+fn combine_entries(
+    base_entry: &HistoricalEntryRaw,
+    quote_entry: &HistoricalEntryRaw,
+    converted_price: BigDecimal,
+) -> Result<HistoricalEntryRaw, InfraError> {
+    let min_timestamp = std::cmp::max(
```

Comment:
min or max ? should definitely be max so you can change the name here

Reply:
Oops yeah, done